src/lib.rs

```diff
@@ -394,35 +398,98 @@ impl Check {
+                                )?),
+                            ))
                         }
                     })
                     .collect()
             }
         };
-        let success = all_outcomes
-            .into_iter()
-            .fold_ok(true, std::ops::BitAnd::bitand)?;
+        let crate_reports: BTreeMap<String, CrateReport> = {
+            let mut reports = BTreeMap::new();
+            for outcome in all_outcomes {
+                let (name, outcome) = outcome?;
+                if let Some(outcome) = outcome {
+                    reports.insert(name, outcome);
+                }
+            }
+            reports
+        };
 
-        Ok(Report { success })
+        Ok(Report { crate_reports })
     }
 }
 
+/// Report of semver check of one crate.
+#[non_exhaustive]
+#[derive(Debug)]
+pub struct CrateReport {
+    /// Bump between the current version and the baseline one.
+    detected_bump: ActualSemverUpdate,
+    /// Minimum bump required to respect semver.
+    /// For example, if the crate contains breaking changes, this is [`ReleaseType::Major`].
+    /// If no bump is required, this is [`Option::None`].
+    required_bump: Option<ReleaseType>,
+}
+
+impl CrateReport {
+    /// Check if the semver check was successful.
+    /// `true` if required bump <= detected bump.
+    pub fn success(&self) -> bool {
+        match self.required_bump {
+            None => true,
+            Some(required_bump) => {
+                match self.detected_bump {
+                    // If user bumped the major version, any breaking change is accepted.
+                    ActualSemverUpdate::Major => true,
+                    ActualSemverUpdate::Minor => {
+                        matches!(required_bump, ReleaseType::Minor | ReleaseType::Patch)
+                    }
+                    ActualSemverUpdate::Patch | ActualSemverUpdate::NotChanged => {
+                        required_bump == ReleaseType::Patch
+                    }
+                }
+            }
+        }
+    }
+
+    /// Minimum bump required to respect semver.
+    /// It's [`Option::None`] if no bump is required.
+    pub fn required_bump(&self) -> Option<ReleaseType> {
+        self.required_bump
+    }
+
+    /// Bump between the current version and the baseline one.
+    pub fn detected_bump(&self) -> ActualSemverUpdate {
+        self.detected_bump
+    }
+}
+
+/// Report of the whole analysis.
+/// Contains a report for each crate checked.
 #[non_exhaustive]
 #[derive(Debug)]
 pub struct Report {
-    success: bool,
+    /// Collection containing the name and the report of each crate checked.
+    crate_reports: BTreeMap<String, CrateReport>,
 }
 
 impl Report {
+    /// `true` if none of the crates violate semver.
     pub fn success(&self) -> bool {
-        self.success
+        self.crate_reports.values().all(|report| report.success())
+    }
+
+    pub fn crate_reports(&self) -> &BTreeMap<String, CrateReport> {
```

Comment:
Please add a doc comment here and in general on all new `pub` items.

Reply:
done in cf47884